### PR TITLE
Fixed issues with the ressource loadind and writing

### PR DESCRIPTION
**Pull request by MrTroble**
Fixed errors with byte offsets
Fixed dereferencing errors
Fixed creation
Removed unused

### CODE REVIEW

File: ShaderTool/Command/Actor.cs [hunk 1]
```diff
@@ -3,25 +3,33 @@
...
+        public float[] vertices;
+        public string  materialName; // will be used to iterate over the materials list and then assign a byte index when making
+        public uint    indexCount;
+        public uint    vertexCount;
+        public byte    layerId;
     }
 
     class Actor {
 
+        public static float[] IDENTETYMATRIX = {
```
Comment (IceDynamix) [inline ShaderTool/Command/Actor.cs]:
```suggestion
        public static float[] IDENTITY_MATRIX = {
```

File: ShaderTool/Command/Map.cs [hunk 2]
```diff
@@ -390,14 +390,13 @@ class Map {
                 ActorData actorData = JsonConvert.DeserializeObject<ActorData>(File.ReadAllText(actorFilePath));
 
                 // Write size of this block in bytes to file
-                // 2 + 2 * (4 + 8) = 26
-                uint actorDataSize = 26 + (actorData.indexCount + actorData.vertexCount) * 4;
+                // 4*16 + 2 + 2*4 = 74
+                uint actorDataSize = 74 + (uint)(actorData.indexCount + actorData.vertices.Length) * 4u;
```
Comment (IceDynamix) [inline ShaderTool/Command/Map.cs]: honestly, having specific hardcoded numbers is not good programming practice and both of us know it, so i'd say we should make the 74 a static class constant in the actor class

File: TGEngine/io/Resource.cpp [hunk 3]
```diff
@@ -78,27 +88,50 @@ void loadResourceFile(const char* name, Map* map) {
 
 		// Reads the indices from the file
 		actorInfo.pIndices = new uint32_t[actorInfo.indexCount]; // Object lifetime ?
-		fread(&actorInfo.pIndices, sizeof(uint32_t), actorInfo.indexCount, file);
+		fread(actorInfo.pIndices, sizeof(uint32_t), actorInfo.indexCount, file);
 
 		// calculates the length of the vertices to read
-		uint32_t len = blocklength - (sizeof(ActorInputInfo) + sizeof(uint32_t) * actorInfo.indexCount);
+		uint32_t len = blocklength - (74 + sizeof(uint32_t) * actorInfo.indexCount);
```
Comment (IceDynamix) [inline TGEngine/io/Resource.cpp]: > honestly, having specific hardcoded numbers is not good programming practice and both of us know it, so i'd say we should make the 74 ~~static class constant in the actor class~~ constant defined somewhere

File: TGEngine/io/Resource.hpp [hunk 4]
```diff
@@ -13,4 +13,10 @@ struct Map {
 	std::vector<tge::gmc::ActorInputInfo> actors;
 };
 
+/*
+ * This loads a Map from a ressource file
```
Comment (IceDynamix) [inline TGEngine/io/Resource.hpp]:
```suggestion
 * This loads a map from a resource file
```

File: TGEngine/io/Resource.cpp (hunk 3 above)
Comment (MrTroble) [reply TGEngine/io/Resource.cpp]: I agree with that tho, I am proposing a Macro

File: ShaderTool/Command/Map.cs (hunk 2 above)
Comment (MrTroble) [reply ShaderTool/Command/Map.cs]: (S.O)

File: TGEngine/io/Resource.cpp (hunk 3 above)
Comment (MrTroble) [reply TGEngine/io/Resource.cpp]: I have an even better idea: We can just make this whole calculation obsolete

**Pull request merged**